code/modules/ai/ai_node.dm

```diff
@@ -38,7 +55,8 @@
 	weights[identifier][name] = amount
 
 /obj/effect/ai_node/Destroy()
-	GLOB.allnodes -= src
+	GLOB.all_nodes[unique_id + 1] = null
```

Comment:
Would it not be better to start the Id at 1 or is this a rust consideration?

Reply:
rust consideration